homeassistant/components/tibber/sensor.py

```diff
@@ -45,108 +49,142 @@
 PARALLEL_UPDATES = 0
 SIGNAL_UPDATE_ENTITY = "tibber_rt_update_{}"
 
-RT_SENSOR_MAP = {
-    "averagePower": ["average power", DEVICE_CLASS_POWER, POWER_WATT, None],
-    "power": ["power", DEVICE_CLASS_POWER, POWER_WATT, None],
-    "powerProduction": ["power production", DEVICE_CLASS_POWER, POWER_WATT, None],
-    "minPower": ["min power", DEVICE_CLASS_POWER, POWER_WATT, None],
-    "maxPower": ["max power", DEVICE_CLASS_POWER, POWER_WATT, None],
-    "accumulatedConsumption": [
+
+class ResetType(Enum):
+    """Data reset type."""
+
+    HOURLY = "hourly"
+    DAILY = "daily"
+    NEVER = "never"
+
+
+@dataclass
+class TibberSensorMetadata:
+    """Metadata for an individual Tibber sensor."""
+
+    name: str
+    device_class: str
+    unit: str | None = None
+    state_class: str | None = None
+    reset_type: ResetType | None = None
+
+
+RT_SENSOR_MAP: dict[str, TibberSensorMetadata] = {
+    "averagePower": TibberSensorMetadata(
+        "average power", DEVICE_CLASS_POWER, POWER_WATT
+    ),
+    "power": TibberSensorMetadata(
+        "power",
+        DEVICE_CLASS_POWER,
+        POWER_WATT,
+    ),
```

Comment:
Especially for fields with default values, I would suggest to use keyword arguments. I.e. either for all or at least `unit`, `state_class`, `reset_type`. This will also help if you ever add any additional fields.

Reply:
Another idea. Maybe add keywords for everything except the `name`. That might also work for other cases. Just so we get at least a bit consistency there.